jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/StepsRunner.java

```diff
@@ -234,35 +232,35 @@ private void authenticatePush() {
   }
 
   private void saveDocker() {
-    ProgressEventDispatcher.Factory childProgressDispatcherFactory =
-        Verify.verifyNotNull(rootProgressDispatcher).newChildProducer();
-
     Optional<DockerClient> dockerClient =
         buildConfiguration.getBaseImageConfiguration().getDockerClient();
     Preconditions.checkArgument(dockerClient.isPresent());
-
-    results.tarPath =
+    ProgressEventDispatcher.Factory childProgressDispatcherFactory =
+        Verify.verifyNotNull(rootProgressDispatcher).newChildProducer();
+    getImageAndLayers(
         executorService.submit(
-            new SaveDockerStep(
+            LocalBaseImageSteps.retrieveDockerDaemonImageStep(
+                executorService,
                 buildConfiguration,
-                dockerClient.get(),
                 childProgressDispatcherFactory,
-                tempDirectoryProvider));
+                dockerClient.get())));
   }
 
   private void extractTar() {
+    Optional<Path> tarPath = buildConfiguration.getBaseImageConfiguration().getTarPath();
+    Preconditions.checkArgument(tarPath.isPresent());
     ProgressEventDispatcher.Factory childProgressDispatcherFactory =
         Verify.verifyNotNull(rootProgressDispatcher).newChildProducer();
-    Future<LocalImage> localImageFuture =
+    getImageAndLayers(
         executorService.submit(
-            () ->
-                new ExtractTarStep(
-                        executorService,
-                        buildConfiguration,
-                        results.tarPath.get(),
-                        childProgressDispatcherFactory,
-                        tempDirectoryProvider)
-                    .call());
+            LocalBaseImageSteps.retrieveTarImageStep(
+                executorService,
+                buildConfiguration,
+                childProgressDispatcherFactory,
+                tarPath.get())));
+  }
+
+  private void getImageAndLayers(Future<LocalImage> localImageFuture) {
```

Comment:
I'm trying to think of a better name for this helper method, since this sounds too general while is only for the base image. And what it does is simple retrieval. `assignLocalImageResult`?